### DISCUSSION TITLE
unable to fetch generated URL with standard protection

### DISCUSSION THREAD

**Discussion by tomotachi** [vercel/community#7017] [Help] [answered]
### Summary

I have the api deployed successfully which works when I access it via browser or when Vercel Authentication is disabled, but it throws 401 when "Standard Protection" is enabled.

![vercel_auth](https://github.com/vercel/community/assets/8308563/c8a16da1-339e-46ff-8adb-20106601d444)

### Steps to Reproduce

The following code prints "Vercel token is valid" but unable to print the returned json from API. 
It could return the json if Vercel Authentication is disabled. Could anyone help please?

`
import requests

def access_vercel_api(token, team_id, api_url):
    # Headers with the authorization token
    headers = {
        "Authorization": f"Bearer {token}"
    }
    
    # Verify the token by making a request to the Vercel API (optional)
    vercel_url = f"https://api.vercel.com/v6/deployments?teamId={team_id}"
    vercel_response = requests.get(vercel_url, headers=headers)
    
    if vercel_response.status_code == 200:
        print("Vercel token is valid.")
        
        # Fetch data from your Vercel API endpoint
        response = requests.get(api_url, headers=headers)
        
        if response.status_code == 200:
            print("API request successful.")
            data = response.json()
            print("Data from API:")
            print(data)
            return data
        else:
            print(f"Failed to fetch data with status code: {response.status_code}")
            print(f"Response: {response.text}")
            return None
    else:
        print(f"Vercel token verification failed with status code: {vercel_response.status_code}")
        print(f"Response: {vercel_response.text}")
        return None

token = '<token>'
team_id = '<team_id>'
api_url = '<generated_api_return_json>'

data = access_vercel_api(token, team_id, api_url)
`

**Comment by QuiiBz** [comment]
Hi, looks like you're passing the bypass secret into an `Authorization` header, but it needs to be set in a `x-vercel-protection-bypass` header instead. See the documentation here: https://vercel.com/docs/security/deployment-protection/methods-to-bypass-deployment-protection/protection-bypass-automation#using-protection-bypass-for-automation

**Reply by tomotachi** [reply]
Checked that "Protection bypass automation" only available for enterprise or pro plan. However, I am using the hobby plan.
I want my react front-end access the standard protected API, both run on Vercel, would there be a way to do with the hobby plan?

**Reply by QuiiBz** [reply, marked as answer]
You can either disable the Vercel Authentication completely on this project, or only enable it on preview deployments. That way, your product deployment will be publicly accessible.

**Reply by tomotachi** [reply]
Thank you. I will disable the Vercel Authentication.

**Reply by QuiiBz** [reply]
Quick update: we just made an update to allow using `x-vercel-protection-bypass` for free, on all plans: https://vercel.com/changelog/improved-security-with-automation-testing-now-available-on-all-plans

**Reply by tomotachi** [reply]
Thanks for your update. 
I have found that if I enabled both the "Vercel Authentication" and "Protection Bypass for Automation", the following error prompts:

Access to fetch at '<myapi>' from origin '<my_react.js>' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

However, if I disable "Vercel Authentication" or disable both, it works.


My code below
`        const response = await fetch('https://<myapi>', {
          method: 'GET',
          headers: {
            'x-vercel-protection-bypass': <my_bypass_secret>
          },
        });
        const data = await response.json();
        setItems(data);
`
